flycheck-idris.el

```diff
@@ -0,0 +1,101 @@
+;;; flycheck-idris.el --- Major mode for editing Idris code -*- lexical-binding: t -*-
+
+;; Copyright (C) 2022
+
+;; Author:
+;; URL: https://github.com/idris-hackers/idris-mode
+;; Keywords: languages
+;; Package-Requires: ((emacs "24") (prop-menu "0.1") (cl-lib "0.5"))
+;; Version: 1.1.0
+
+
+;;; Commentary:
+
+;; FlyCheck checkers for Idris(2)
+
+;;; Code:
+
+(require 'flycheck)
+(require 'idris-mode)
+
+(flycheck-define-checker idris
+  "An Idris syntax and type checker."
+  :command ("idris"
+            "--check" "--nocolor" "--warnpartial"
+            ;; Compute the command-line options similarly to inferior-idris
+            (eval (idris-compute-flags))
+            source-original)
+  :error-patterns
+  ((warning line-start
+            (file-name)
+            ":"
+            line
+            ":"
+            column
+            "-"
+            end-column
```

Comment:
The original checker in idris-mode.el was missing `end-column` in the error pattern definition
and also the pattern for message did not work well.